pkg/scheduler/core/equivalence/eqivalence.go

```diff
@@ -150,6 +140,25 @@ func (c *Cache) InvalidateCachedPredicateItemForPodAdd(pod *v1.Pod, nodeName str
 		}
 	}
 	c.InvalidatePredicatesOnNode(nodeName, invalidPredicates)
+
+	// MatchInterPodAffinity: we assume scheduler can make sure newly bound pod
+	// will not break the existing inter pod affinity. So normally we does not need to
+	// invalidate MatchInterPodAffinity when pod added.
+	//
+	// NOTE: assumptions above will not stand when we implemented features like
+	// RequiredDuringSchedulingRequiredDuringExecutioc.
+	//
+	// But the exception is: pod has anti-affinity to itself. This is common in a ReplicaSet,
+	// so we have to invalidate MatachInterPodAffinity in this case.
+	//
+	// TODO(harry): This is aggressive. We should consider a more fine-grained way later,
+	// especially after eCache is merged to predicates pkg (#67241). Then we can do invalidation:
+	// 1. only for nodes in the same failure domain;
```

Comment:
failure domain != topology

I don't know if we can put a bound on the "blast radius" of affinity and anti-affinity rules.